### PR TITLE
Add Port Annotation Workflow. Closes #7

### LINKED ISSUE #7: Annotating Ports
It might be nice to extend the current annotations API for labeling ports as well. This should preferably happen in the server itself just to maintain consistency with the current calculations.

### PR DESCRIPTION
**Pull request by umesh-timalsina**
This PR intends to modify the `TagSelector` widget, currently used for `annotorious` to add capabilities for annotating ports. Currently, I used the example `all-collpitts.json` file to annotate. The workflow check list:

- [x] Select a Component region using the rectangle select tool
- [x] label it as a proper Component
- [x] browser sends the image and annotation to the server to detect the likely ports
- ~[x] browser requests the expected port names for the Component.~
- [x] Expected ports are shown to the user in the widget and detected ports are selected with a greyed out rectangle
- [x] User selects the ports (likely just clicking on the detected ones) and selects the port name

PS: Currently, a bug exists in [`annotorious`](https://github.com/recogito/annotorious) such that we have to keep a running list of annotations. This will be fixed in the next release of the tool. See https://github.com/recogito/annotorious-openseadragon/issues/54 for further details.

### _Edit 1_:Port Annotation Deletion and Manual Port Annotation?
Currently, I assume that the Port annotation capability is only available in server. However, should we allow manual port annotation to the user? The workflow for that is not clear at this point because the user cannot select parent annotation at this point. This should probably be tackled in another issue/pr set, but what if the user deletes a port annotation but realized that he/she needs it again? Should we re-request port annotations from the server?

### CODE REVIEW

File: js/circuit-annotator/src/CircuitComponents.json [hunk 1]
```diff
@@ -334,6 +334,10 @@
         "p",
         "n"
       ]
+    },
+    "Port": {
```
Comment (brollb) [inline js/circuit-annotator/src/CircuitComponents.json:338]: What is this label used for? Subcircuits?

**Review by brollb — COMMENTED** [review]
Nice. It is nice to see that it is using the expected port names to update the tag viewer. A couple questions/comments:
- I noticed that deleting a component doesn't remove the ports for the given component.
- How can you add port labels for a new component?

File: js/circuit-annotator/src/CircuitComponents.json (hunk 1 above)
Comment (umesh-timalsina) [reply js/circuit-annotator/src/CircuitComponents.json:338]: With the new hierarchical/flat annotation conversion from the backend this is unnecessary now.

**Comment by umesh-timalsina** [comment]
>I noticed that deleting a component doesn't remove the ports for the given component.
How can you add port labels for a new component?

We should probably add removal in the handler here:
https://github.com/symbench/scm/blob/b9160a8d6e4d25502908151c77745eed62919cd3/js/circuit-annotator/src/Components/OpenSeadragon/OpenSeadragon.svelte#L96

**Comment by umesh-timalsina** [comment]
> How can you add port labels for a new component?

It should work out of the box. If no port labels are available, then you have to manually type it in. We could add some stricter validation for ports as well, but any label provided to a Port is valid for now.

**Comment by brollb** [comment]
Perhaps I am missing the flow that you are imagining. I am clicking on a component, selecting "Capacitor", selecting "OK", and then nothing is happening. When do I select the ports?

**Comment by umesh-timalsina** [comment]
> Perhaps I am missing the flow that you are imagining. I am clicking on a component, selecting "Capacitor", selecting "OK", and then nothing is happening. When do I select the ports?

When you select a component, I am imaging the following REST API endpoint:
1. `POST /annotate/ports - body {annotation}`. This will give a list of probable port annotations and the frontend code will work as expected.

**Comment by umesh-timalsina** [comment]
https://github.com/symbench/scm/blob/231510e8d44530e4c95907cdbec0ba8518b3cce3/js/circuit-annotator/src/Components/OpenSeadragon/AnnotoriousPortsAnnotatorPlugin.js#L4-L28

**Pull request merged**